nixos/modules/services/web-apps/lasuite-docs.nix

```
@@ -0,0 +1,473 @@
+    ;
+
+  cfg = config.services.lasuite-docs;
+
+  gunicornSettings = pkgs.writeText "gunicorn-settings.py" ''
+    bind = [ "${cfg.listenAddress}:${toString cfg.port}" ]
+    name = "impress"
+    python_path = "${cfg.package}/lib/docs/src/"
+
+    graceful_timeout = 90
+    timeout = 90
+    workers = 3
+
+    accesslog = "-"
+    errorlog = "-"
+    loglevel = "debug"
+  '';
+
+  commonServiceConfig = {
+    RuntimeDirectory = "lasuite-docs";
+    StateDirectory = "lasuite-docs";
+    WorkingDirectory = "/var/lib/lasuite-docs";
+
+    User = "lasuite-docs";
+    DynamicUser = true;
+    SupplementaryGroups = mkIf cfg.redis.createLocally [ "redis-lasuite-docs" ];
+    # hardening
+    AmbientCapabilities = "";
+    CapabilityBoundingSet = [ "" ];
+    DevicePolicy = "closed";
+    LockPersonality = true;
+    # MemoryDenyWriteExecute = true;
+    NoNewPrivileges = true;
+    PrivateDevices = true;
+    PrivateTmp = true;
+    PrivateUsers = true;
+    ProcSubset = "pid";
+    ProtectClock = true;
+    ProtectControlGroups = true;
+    ProtectHome = true;
+    ProtectHostname = true;
+    ProtectKernelLogs = true;
+    ProtectKernelModules = true;
+    ProtectKernelTunables = true;
+    ProtectProc = "invisible";
+    ProtectSystem = "strict";
+    RemoveIPC = true;
+    RestrictAddressFamilies = [ "AF_INET AF_INET6 AF_UNIX" ];
+    RestrictNamespaces = true;
+    RestrictRealtime = true;
+    RestrictSUIDSGID = true;
+    SystemCallArchitectures = "native";
+    UMask = "0077";
+  };
+
+  pythonPreloadSecrets = ''
+    ${
+      if cfg.secretKeyPath != null then
+        "export DJANGO_SECRET_KEY=$(cat ${cfg.secretKeyPath})"
+      else
+        ''
+          if [[ ! -f /var/lib/lasuite-docs/django_secret_key ]]; then
+            (
+              umask 0377
+              tr -dc A-Za-z0-9 < /dev/urandom | head -c64 | ${pkgs.moreutils}/bin/sponge /var/lib/lasuite-docs/django_secret_key
+            )
+          fi
+          export DJANGO_SECRET_KEY=$(cat /var/lib/lasuite-docs/django_secret_key)
+        ''
+    }
+    ${optionalString (
+      cfg.s3.accessKeyIDPath != null
+    ) "export AWS_S3_ACCESS_KEY_ID=$(cat ${cfg.s3.accessKeyIDPath})"}
+    ${optionalString (
+      cfg.s3.secretAccessKeyPath != null
+    ) "export AWS_S3_SECRET_ACCESS_KEY=$(cat ${cfg.s3.secretAccessKeyPath})"}
+    ${optionalString (
+      cfg.oidc.clientSecretPath != null
+    ) "export OIDC_RP_CLIENT_SECRET=$(cat ${cfg.oidc.clientSecretPath})"}
+    ${optionalString (
+      cfg.collaborationServer.serverSecretPath != null
+    ) "export COLLABORATION_SERVER_SECRET=$(cat ${cfg.collaborationServer.serverSecretPath})"}
+    ${optionalString (
+      cfg.collaborationServer.yproviderApiKeyPath != null
+    ) "export Y_PROVIDER_API_KEY=$(cat ${cfg.collaborationServer.yproviderApiKeyPath})"}
+  '';
+in
+{
+  options.services.lasuite-docs = {
+    enable = mkEnableOption "SuiteNumérique Docs";
+
+    package = mkPackageOption pkgs "lasuite-docs" { };
```

```suggestion
    backendPackage = mkPackageOption pkgs "lasuite-docs" { };
```

For consistency, as it is unclear what `package` would refer to.